src/pages/home/report/ReportActionContextMenuItem.js

```diff
@@ -29,52 +35,107 @@ function getButtonState(isHovered = false, isPressed = false) {
 const propTypes = {
     icon: PropTypes.elementType.isRequired,
     text: PropTypes.string.isRequired,
+    successIcon: PropTypes.elementType,
+    successText: PropTypes.string,
     isMini: PropTypes.bool,
+    onPress: PropTypes.func.isRequired,
 };
 
 const defaultProps = {
     isMini: false,
+    successIcon: null,
+    successText: '',
 };
 
-const ReportActionContextMenuItem = (props) => {
-    const {getButtonStyle, getIconFillColor} = getReportActionContextMenuItemStyles(props.isMini);
-    return (
-        props.isMini
-            ? (
-                <Tooltip text={props.text}>
-                    <Pressable style={({hovered, pressed}) => getButtonStyle(getButtonState(hovered, pressed))}>
+class ReportActionContextMenuItem extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            hasRun: false,
```

Comment:
Maybe let's just call this state variable `success` instead of `hasRun`

Reply:
Ok.